src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2912,6 +2912,15 @@ def transform(dataset):
         Ref('encryption_options'),
         Bool('encryption', default=False),
         Bool('inherit_encryption', default=True),
+        List(
```

Comment:
It's pretty unobvious API. The natural use case here is to replace dataset's user properties with the ones provided. If you want to avoid re-sending properties every time, add a second field `user_properties_update` into `do_update` method.